proxy/http/HttpSM.cc

```diff
@@ -751,6 +751,25 @@ HttpSM::state_read_client_request_header(int event, void *data)
     milestones[TS_MILESTONE_UA_READ_HEADER_DONE] = Thread::get_hrtime();
   }
 
+#if TS_HAS_TLS_EARLY_DATA
+  SSLNetVConnection *ssl_vc = dynamic_cast<SSLNetVConnection *>(netvc);
+  if (ssl_vc != nullptr && ssl_vc->read_from_early_data) {
+    ssl_vc->read_from_early_data = false;
+    // Only allow early data for GET and HEAD methods.
```

Comment:
This is allows for more methods than GET and HEAD.

Reply:
fixed